tests/tests/test_unit/test_regions/test_region_analysis.py

```diff
@@ -0,0 +1,89 @@
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import pytest
+import tifffile
+from napari.layers import Labels
+
+from brainreg_segment.regions.analysis import (
+    check_list_only_nones,
+    summarise_brain_regions,
+    summarise_single_brain_region,
+)
+
+region_image_path = Path(
+    "tests/data/brainreg_output/manual_segmentation/"
+    "sample_space/regions/region_0.tiff"
+)
+atlas_resolution = (50, 50, 50)
+label_name = "test_label"
+
+
+@pytest.fixture
+def region_image():
+    return tifffile.imread(region_image_path)
+
+
+@pytest.fixture
+def labels_layer(region_image):
+    labels = Labels(region_image, name=label_name)
+    return labels
+
+
+@pytest.fixture
+def empty_labels_layer(region_image):
+    labels = Labels(np.zeros_like(region_image))
+    return labels
+
+
+def test_check_list_only_ones():
+    assert check_list_only_nones([None, None, None]) is True
+    assert check_list_only_nones([None, 1, None]) is False
+    assert check_list_only_nones([1, "two", 3]) is False
+
+
+def test_summarise_brain_regions(labels_layer, empty_labels_layer, tmp_path):
+    filename = tmp_path / "summary.csv"
+
+    # check correct behaviour if all regions are not empty
+
+    summarise_brain_regions(
+        [labels_layer, labels_layer, labels_layer], filename, atlas_resolution
+    )
+    df = pd.read_csv(filename)
+    assert df.shape == (3, 11)
+    assert df["region"].iloc[0] == label_name
+    assert np.isclose(df["volume_mm3"].iloc[0], 2.23025)
```

Comment:
I wonder whether it's worth noting (not sure where though? maybe in a comment next to them or at the top of the file?) that these are (presumably) values we've remembered from previous analysis and a regression testing against?